pkg/entities/nfts_test.go

```diff
@@ -762,3 +765,330 @@ func TestEntity_GetNFTDetail(t *testing.T) {
 		})
 	}
 }
+
+func TestEntity_CreateNFTCollection(t *testing.T) {
+	type fields struct {
+		repo        *repo.Repo
+		store       repo.Store
+		log         logger.Logger
+		dcwallet    discordwallet.IDiscordWallet
+		discord     *discordgo.Session
+		cache       cache.Cache
+		svc         *service.Service
+		cfg         config.Config
+		indexer     indexer.Service
+		abi         abi.Service
+		marketplace marketplace.Service
+	}
+	type args struct {
+		req request.CreateNFTCollectionRequest
+	}
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	cfg := config.Config{
+		DBUser: "postgres",
+		DBPass: "postgres",
+		DBHost: "localhost",
+		DBPort: "5434",
+		DBName: "mochi_local",
+
+		InDiscordWalletMnemonic: "holiday frequent toy bachelor auto use style result recycle crumble glue blouse",
+		FantomRPC:               "sample",
+		FantomScan:              "sample",
+		FantomScanAPIKey:        "sample",
+
+		EthereumRPC:        "sample",
+		EthereumScan:       "sample",
+		EthereumScanAPIKey: "sample",
+
+		BscRPC:        "sample",
+		BscScan:       "sample",
+		BscScanAPIKey: "sample",
+
+		DiscordToken: "sample",
+
+		RedisURL: "redis://localhost:6379/0",
+	}
+
+	s := pg.NewPostgresStore(&cfg)
+	r := pg.NewRepo(s.DB())
+
+	nftCollection := mock_nft_collection.NewMockStore(ctrl)
+	mockIndexer := mock_indexer.NewMockService(ctrl)
+	mockAbi := mock_abi.NewMockService(ctrl)
+	mockMarketplace := mock_marketplace.NewMockService(ctrl)
+
+	r.NFTCollection = nftCollection
+
+	tests := []struct {
+		name              string
+		fields            fields
+		args              args
+		wantNftCollection *model.NFTCollection
+		wantErr           bool
+	}{
+		// TODO: Add test cases.
```

Comment:
add more test case for error, toggle test coverage to see if you miss any error case